plugins/modules/source_control/gitlab/gitlab_integration.py

```diff
@@ -0,0 +1,509 @@
+    default: ["push", "issues", "confidential_issues", "merge_requests", "tag_push", "note", "confidential_note", "job", "pipeline", "wiki_page"]
+    type: list
+    elements: str
+  state:
+    description:
+      - Create or delete an integration.
+      - Possible values are C(present) and C(absent).
+    default: "present"
+    choices: ["present", "absent"]
+    type: str
+'''
+
+EXAMPLES = '''
+# Setup email on push for this project
+- name: Email me on push
+  community.general.gitlab_integration:
+    api_url: https://gitlab.com
+    api_token: foobar
+    project: 123456
+    integration: emails-on-push
+    params:
+      recipients: foo@example.com
+      disable_diffs: true
+    state: present
+  delegate_to: localhost
+
+# This will always be set to change because a non-null token is mandatory
+- name: Trigger packagist update on push events (only)
+  community.general.gitlab_integration:
+    api_url: https://gitlab.com
+    api_token: foobar
+    project: foo/proj
+    integration: packagist
+    events: ["push"]
+    params:
+      username: foo
+      token: bar
+      server: https://packagist.org
+  no_log: True
+  delegate_to: localhost
+
+- Idempotency is only partially provided since GitLab does
+  not expose secret parameters like tokens or passwords.
+  See U(https://gitlab.com/gitlab-org/gitlab-ce/issues/46313)
+'''
+
+RETURN = '''
+---
+integration:
+  description: A dict containing key/value pairs representing GitLab integration
+  returned: success
+  type: dict
+  sample:
+    id: 40812345
+    push_events: true
+    issues_events: true
+    confidential_issues_events: true
+    merge_requests_events: true
+    tag_push_events: true
+    note_events: true
+    confidential_note_events: true
+    job_events: true
+    pipeline_events: true
+    wiki_page_events: true
+    recipients: me@example.com
+    disable_diffs: null
+    send_from_committer_email: null
+    title: Emails on push
+    created_at: '2018-05-13T03:08:07.943Z'
+    updated_at: '2018-05-13T03:09:17.651Z'
+    active: true
+    properties:
+      send_from_committer_email: null
+      disable_diffs: null
+      recipients: me@example.com
+    project_id: 1234567
+state:
+  description: A string indicating whether the integration was "created" or "changed"
+  returned: success
+  type: str
+  sample: created
+'''
+
+import traceback
+
+from ansible.module_utils.six import iteritems
+
+GITLAB_IMP_ERR = None
+try:
+    import gitlab
+    HAS_GITLAB_PACKAGE = True
+except ImportError:
+    GITLAB_IMP_ERR = traceback.format_exc()
+    HAS_GITLAB_PACKAGE = False
+
+from ansible.module_utils.api import basic_auth_argument_spec
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils._text import to_native
+
+from ansible_collections.community.general.plugins.module_utils.gitlab import auth_argument_spec, gitlabAuthentication
```

Comment:
```suggestion
from ansible_collections.community.general.plugins.module_utils.gitlab import auth_argument_spec, gitlab_authentication
```
It has been renamed at some point.